Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h

```diff
@@ -444,7 +444,33 @@ struct CLuaFunctionParserBase
         }
         else if constexpr (std::is_same_v<T, CLuaFunctionRef>)
         {
-            return luaM_toref(L, index++);
+        #ifdef MTA_CLIENT
+            CLuaMain* pLuaMain = g_pClientGame->GetLuaManager()->GetVirtualMachine(L);
+        #else
+            CLuaMain* pLuaMain = g_pGame->GetLuaManager()->GetVirtualMachine(L);
+        #endif
+            const void* pFuncPtr = lua_topointer(L, index);
+
+            if (CRefInfo* pInfo = MapFind(pLuaMain->m_CallbackTable, pFuncPtr))
+            {
+                // Re-use the lua ref we already have to this function
+                pInfo->ulUseCount++;
+                ++index;
```

Comment:
This is repeated in the `else` branch below. It could be moved to before the `if`.

Reply:
The else branch uses index pre-increment, so we can't move that up (at least without subtracting 1)